### PR TITLE
Add configurable Insecure mode

### PR DESCRIPTION
**Pull request by TheColonelK**
Allow the user to configure Insecure Mode

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -59,12 +61,16 @@ const (
 // To create a new client, please provide the url of the flyte server and the timeout.
 // timeout specifies a time limit for requests made by this
 // client. A timeout of zero means no timeout.
```
Comment (ukjasonwright) [inline client/client.go:62]: might wanna update this doc

File: client/client_test.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func Test_NewClient_ShouldRetryOnErrorGettingFlyteApiLinks(t *testing.T) {
 	baseUrl, _ := url.Parse(server.URL)
 
 	// when
-	client := NewClient(baseUrl, 10*time.Second)
```
Comment (ukjasonwright) [inline client/client_test.go]: add a space!! "NewClient(baseUrl, 10*time.Second, false)"

also elsewhere

File: client/client.go [hunk 3]
```diff
@@ -59,12 +61,16 @@ const (
...
-func NewClient(rootURL *url.URL, timeout time.Duration) Client {
+func NewClient(rootURL *url.URL, timeout time.Duration, IsInsecure bool) Client {
```
Comment (ukjasonwright) [inline client/client.go]: argument name should be lowercase

File: client/client.go [hunk 4]
```diff
@@ -59,12 +61,16 @@ const (
...
 	baseUrl := getBaseURL(*rootURL)
+
 	client := &client{
 		baseURL: baseUrl,
 		httpClient: &http.Client{
 			Timeout: timeout,
+			Transport:  &http.Transport{
```
Comment (ukjasonwright) [inline client/client.go]: The docs for this state "	// This should be used only for testing." - why are we adding it?

**Review by ukjasonwright — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 3 above)
Comment (TheColonelK) [reply client/client.go]: Done

File: client/client.go (hunk 4 above)
Comment (TheColonelK) [reply client/client.go]: They may say that but it's used alot not just for testing

File: client/client_test.go (hunk 2 above)
Comment (TheColonelK) [reply client/client_test.go]: Done

File: client/client.go (hunk 1 above)
Comment (TheColonelK) [reply client/client.go:62]: Done

**Pull request closed without merging**